src/integer_mod_q/mat_zq/solve.rs

```diff
@@ -124,55 +391,220 @@ fn find_invertible_entry_column(
+
+        assert_eq!(y, mat * x)
+    }
+
+    /// Ensure that a solution is found, even if a matrix in `solve_hensel` has
+    /// rows containing only zeros after gaussian elimination.
+    #[test]
+    #[ignore]
+    fn solution_zero_rows() {
+        let mat = MatZq::from_str("[[6,1],[3,1]] mod 36").unwrap();
+        let y = MatZq::from_str("[[6],[3]] mod 36").unwrap();
+
+        let x = mat.solve_gaussian_elimination(&y).unwrap();
+
+        let cmp_sol = MatZq::from_str("[[1],[0]] mod 6").unwrap();
+        assert_eq!(cmp_sol, x);
+    }
+
+    /// Ensure that a solution is found with prime modulus and more rows than columns
+    /// (This test does not work with the current implementation).
+    #[test]
+    #[ignore]
+    fn solution_more_rows() {
+        let mat = MatZq::from_str("[[5,6],[11, 12],[11, 12]] mod 20").unwrap();
+        let y = MatZq::from_str("[[5],[2],[2]] mod 20").unwrap();
+
+        let x = mat.solve_gaussian_elimination(&y).unwrap();
+
+        let cmp_sol = MatZq::from_str("[[5],[1]] mod 20").unwrap();
+        assert_eq!(cmp_sol, x);
+    }
+
+    /// Ensure that a solution is found in random matrices (for testing purposes).
+    #[test]
+    #[ignore]
+    fn random_matrix_modulus() {
+        // modulus: 2:100,      rows: 1:10,     columns: 1:10    => <7% false Nones
+        // modulus: 2:10000,    rows: 1:10,     columns: 1:10    => <7% false Nones
+        // modulus: 2:10000,    rows: 1:10,     columns: 10:20   => <0.5% false Nones
+        // modulus: 2:10000,    rows: 50:100,   columns: 100:200 => not measurable
+
+        let mut none_count = 0;
+        let mut correct_count = 0;
+        let mut false_count = 0;
+
+        for i in 0..1000 {
+            let modulus_sample = Z::sample_uniform(2, 10000).unwrap();
+            let row_sample = &Z::sample_uniform(50, 100).unwrap();
+            let column_sample = &Z::sample_uniform(100, 200).unwrap();
+
+            let modulus = Modulus::from(modulus_sample);
+            let mat = MatZq::sample_uniform(row_sample, column_sample, &modulus);
+            let x = MatZq::sample_uniform(column_sample, 1, &modulus);
+            let y = &mat * &x;
+
+            if let Some(solution) = mat.solve_gaussian_elimination(&y) {
+                if &mat * &solution == y {
+                    correct_count += 1;
+                    println!("{}: Correct!", i);
+                } else {
+                    false_count += 1;
+                    println!("{}: False", i);
+                    println!("\t Matrix: {} \n\t y: {} \n\t x: {}", &mat, y, &x);
+                }
+            } else {
+                none_count += 1;
+                println!("{}: None", i);
+                println!("\t Matrix: {} \n\t y: {} \n\t x: {}", mat, y, x);
+            }
+        }
+
+        println!("Nones: {}", none_count);
+        println!("Corrects: {}", correct_count);
+        println!("False: {}", false_count);
+    }
```

Comment:
do you want this test to remain in the code as it is?

Reply:
I am not sure. But I thought that it does not hurt to leave it like that.
Or do you think I should remove it or change it?